packages/core/android/src/main/kotlin/com/datadog/reactnative/DdSdkConfigurationExt.kt

```diff
@@ -75,31 +76,108 @@ internal fun ReadableMap.asProxyConfig(): Pair<Proxy, ProxyAuthenticator?>? {
         port = getInt("port")
     }
 
-    val type =
-        getString("type")?.let {
-            when (it.lowercase(Locale.US)) {
+    val type = getString("type")
+
+    if (address != null && port != null && type != null) {
+        return JSONProxyConfiguration(
+            type,
+            address,
+            port,
+            getString("username"),
+            getString("password")
+        ).asProxyConfig()
+    } else {
+        return null
+    }
+}
+
+@Suppress("LabeledExpression")
+internal fun ReadableArray.asFirstPartyHosts(): Map<String, Set<TracingHeaderType>> {
+    val firstPartyHosts = this.toArrayList() as List<ReadableMap>
+    return firstPartyHosts.mapNotNull<ReadableMap, JSONFirstPartyHost> {
+        it.getString("match")?.let { match ->
+            it.getArray("propagatorTypes")?.let { propagatorTypes ->
+                return@mapNotNull JSONFirstPartyHost(
```

Comment:
I'm curious if you remove the `null` at the end of this, can you remove the `return@mapNotNull`, since `let` should return the result of the lambda (null in cases where it didn't execute)